observability-query.md

```diff
@@ -0,0 +1,199 @@
+---
+title: observability-query-language
+authors:
+  - alanconway
+reviewers:
+  - TBD
+approvers:
+  - TBD
+api-approvers:https://github.com/grafana/tempo/pull/1378
+  - TBD
+creation-date: 2022-05-06
+last-updated: 2022-05-06
+tracking-link:
+  - https://issues.redhat.com/browse/LOG-2580
+see-also:
+replaces:
+superseded-by:
+---
+
+# Observability Query Language
+
+## Summary
+
+Different observability signal types may be stored in different stores, with different query languages.
+In order to correlate different signal types, a "correlation service" must query multiple stores.
+This proposal describes a common query language (ObsQL) that can be mapped to the native query language of each store.
+This is important to separate the handling of different query languages from the logic of correlating signals.
+
+## Motivation
+
+There are several signal types that we would like to be able correlate including logs, metrics, traces and k8s events.
+Each signal type can have a different store, with its own query language. For example:
+
+- Logs: Loki ([LogQL](https://grafana.com/docs/loki/latest/logql/)), Elasticsearch ([ES Query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html))
+- Metrics: Prometheus ([PromQL](https://prometheus.io/docs/prometheus/luerying/basics/))
+- Traces: Tempo ([TraceQL](https://github.com/grafana/tempo/pull/1378)), Jaeger ([NoSQL](https://www.jaegertracing.io/docs/1.33/features/#multiple-storage-backends))
+- K8s Events: K8s API server ([K8s Selectors](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/))
+
+To correlate different signals, a "correlation service" must query multiple stores for multiple signal sets.
+
+A common query language separates correlation logic from the details of handling multiple query languages.
+A translation library from ObsQL to each back-end query language will translate observability queries into native queries.
+
+### User Stories
+
+Some of these stories are for the Openshift developers, this API is important for internal use.
+
+#### The cluster logging team want a common log exploration API for Loki and Elasticsearch
+
+A single API for retrieving logs regardless of the store technology will enable:
+
+- Console views for logs.
+- Programmatic access for users to log data.
+- Direct access to node log files via the k8s API (like `oc logs`) with a consistent API.
```

Comment:
This seems like a valuable idea, but im confused over the querying point of reference? 

- I have an api for querying pod logs directly?
- I have an api for query logs via OpenShift console?
- I have an api for querying all signals and have correlation?